pyuvdata/telescopes.py

```diff
@@ -94,212 +242,570 @@ def __init__(self):
+
+        full_list = astropy_sites_list + known_telescope_list
+        if warn and len(full_list) > 0:
+            warn_str = ", ".join(full_list) + " are not set or are being overwritten. "
+            specific_str = []
+            if len(astropy_sites_list) > 0:
+                specific_str.append(
+                    ", ".join(astropy_sites_list) + " are set using values from "
+                    f"astropy sites for {self.name}."
+                )
+            if len(known_telescope_list) > 0:
+                specific_str.append(
+                    ", ".join(known_telescope_list) + " are set using values "
+                    f"from known telescopes for {self.name}."
+                )
+            warn_str += " ".join(specific_str)
+            warnings.warn(warn_str)
 
-    Parameters
-    ----------
-    telescope_name : str
-        Name of a telescope
-    telescope_dict_in: dict
-        telescope info dict. Default is None, meaning use KNOWN_TELESCOPES
-        (other values are only used for testing)
+        if run_check:
+            self.check(
+                check_extra=check_extra, run_check_acceptability=run_check_acceptability
+            )
 
-    Returns
-    -------
-    Telescope object
-        The Telescope object associated with telescope_name.
-    """
-    if telescope_dict_in is None:
-        telescope_dict_in = KNOWN_TELESCOPES
+    @classmethod
+    def from_known_telescopes(
+        cls,
+        name: str,
+        *,
+        run_check: bool = True,
+        check_extra: bool = True,
+        run_check_acceptability: bool = True,
+        known_telescope_dict: dict = KNOWN_TELESCOPES,
+    ):
+        """
+        Create a new Telescope object using information from known_telescopes.
+
+        Parameters
+        ----------
+        name : str
+            Name of the telescope.
+        run_check : bool
+            Option to check for the existence and proper shapes of parameters.
+        check_extra : bool
+            Option to check optional parameters as well as required ones.
+        run_check_acceptability : bool
+            Option to check acceptable range of the values of parameters.
+        known_telescope_dict: dict
+            This should only be used for testing. This allows passing in a
+            different dict to use in place of the KNOWN_TELESCOPES dict.
+
+        Returns
+        -------
+        Telescope
+            A new Telescope object populated with information from
+            known_telescopes.
+
+        """
+        tel_obj = cls()
+        tel_obj.name = name
+        tel_obj.update_params_from_known_telescopes(
+            warn=False,
+            run_check=run_check,
+            check_extra=check_extra,
+            run_check_acceptability=run_check_acceptability,
+            known_telescope_dict=known_telescope_dict,
+        )
+        return tel_obj
+
+    @classmethod
+    def from_params(
```

Comment:
Should this be called `new` to have a similar name to the analogous methods on UVData and UVCal?

Reply:
I would endorse this -- presumably that means we need to add `initializers.new_telescope` as well?

Reply:
I also would endorse this also.